proxy/InkAPI.cc

```diff
@@ -7748,7 +7748,7 @@ TSHttpTxnServerPush(TSHttpTxn txnp, const char *url, int url_len)
   Http2Stream *stream = dynamic_cast<Http2Stream *>(sm->ua_txn);
   if (stream) {
     Http2ClientSession *parent = static_cast<Http2ClientSession *>(stream->get_parent());
-    if (!parent->is_url_pushed(url, url_len)) {
+    if (parent && !parent->connection_state.is_state_closed() && !parent->is_url_pushed(url, url_len)) {
```

Comment:
Should the mutex of parent or connection_state be locked?

Reply:
yes, good catch!